src/util.cc

```diff
@@ -787,4 +812,49 @@ v8::Maybe<int32_t> GetValidatedFd(Environment* env,
   return v8::Just(static_cast<int32_t>(fd));
 }
 
+v8::Maybe<int> GetValidFileMode(Environment* env,
+                                v8::Local<v8::Value> input,
+                                uv_fs_type type) {
```

Comment:
Why not add this to `node_file.cc` instead? 

And I think this method could be simplified and renamed to something like: 
`bool ValidateFileMode(Environment* env, v8::Local<v8::Value> input, uv_fs_type type, int* flags)`.

I needed to read [v8:Maybe](https://v8docs.nodesource.com/node-20.3/d9/d4b/classv8_1_1_maybe.html#ab9c5335a9d708d82c00b0e7ada488c08) to understand when this will produce a false condition when using `to` method.

Reply:
`GetValidatedFd` is also in `node_util.cc` and if I remember correctly I put it there because of a feedback, so I don't have a strong opinion here.

To answer your questions:

- v8::Maybe is V8 way of writing `std::optional`. std::optional<int> vs `v8::Maybe<int>` are some-what exactly the same thing. More information about maybe types can be found on Node.js documentation: https://github.com/nodejs/node/tree/main/src#maybe-types
- Passing pointer as an argument to mutate is a C way of doing things. This is somewhat similar to what the current implementation is but without C like function declaration. Referencing `src/readme.md` file:

> A typical way of accessing the value is using the .To() function, which returns a boolean indicating success of the operation (i.e. the Maybe not being empty) and taking a pointer to a T to store the value if there is one.

Reply:
If `GetValidatedFd` is written in that way, makes sense to keep the same pattern, in case we want to change that, let's keep it for a future PR.